File: Frontend/src/types/task.ts
export interface TaskDto {
  taskId: string;
  sublistId: string;
  order: number;
  title: string;
  description: string;
  isCompleted: boolean;
  isDeleted: boolean;
  updatedAt: string;
  reminderAt?: string;
  dueTime?: string;
}
export interface Task extends TaskDto {
  isSynced?: boolean;
}

File: Backend/Tazzker.Domain/Task.cs
using System.ComponentModel.DataAnnotations;

namespace Tazzker.Domain
{
    public class Task
    {
        [Key]
        public Guid TaskId { get; set; }
        public Guid? SublistId { get; set; }
        public Guid UserId { get; set; }
        public float Order { get; set; }

        public string Title { get; set; } = null!;
        public string Description { get; set; } = null!;
        public bool IsCompleted { get; set; }
        public bool IsDeleted { get; set; }
        
        public DateTime UpdatedAt { get; set; }
        public DateTime? ReminderAt { get; set; }
        public DateTime? DueTime { get; set; }

    }
}
